components/textfield/index.css

```diff
@@ -532,323 +229,151 @@ governing permissions and limitations under the License.
-		);
-	}
-
 	.is-disabled & {
-		color: var(--spectrum-textfield-text-color-disabled);
+		color: var(--highcontrast-textfield-text-color-disabled, var(--spectrum-textfield-text-color-disabled));
 	}
 }
 
 /********* Child Component - Help Text *********/
 .spectrum-HelpText {
 	.spectrum-Textfield & {
-		margin-block-start: var(
-			--mod-textfield-helptext-spacing-block,
-			var(--spectrum-textfield-helptext-spacing-block)
-		);
+		margin-block-start: var(--mod-textfield-helptext-spacing-block, var(--spectrum-textfield-helptext-spacing-block));
 		grid-row: 3;
 		grid-column: 1 / span 2;
 	}
+
+	.is-disabled & .spectrum-HelpText-text {
+		color: var(--highcontrast-textfield-text-color-disabled, var(--spectrum-textfield-text-color-disabled));
+	}
 }
 
 /********* Child Element - Character Count *********/
 .spectrum-Textfield-characterCount {
+	color: var(--mod-textfield-character-count-color, var(--spectrum-textfield-character-count-color));
 	display: inline-flex;
 	align-items: flex-end;
 	justify-content: flex-end;
 	inline-size: auto;
-	margin-block-end: var(
-		--mod-textfield-character-count-spacing-block,
-		var(--spectrum-textfield-character-count-spacing-block)
-	);
-	margin-inline-start: var(
-		--mod-textfield-character-count-spacing-inline,
-		var(--spectrum-textfield-character-count-spacing-inline)
-	);
+	margin-block-end: var(--mod-textfield-character-count-spacing-block, var(--spectrum-textfield-character-count-spacing-block));
+	margin-inline-start: var(--mod-textfield-character-count-spacing-inline, var(--spectrum-textfield-character-count-spacing-inline));
 	margin-inline-end: 0;
-	font-size: var(
-		--mod-textfield-character-count-font-size,
-		var(--spectrum-textfield-character-count-font-size)
-	);
-	font-family: var(
-		--mod-textfield-character-count-font-family,
-		var(--spectrum-textfield-character-count-font-family)
-	);
-	font-weight: var(
-		--mod-textfield-character-count-font-weight,
-		var(--spectrum-textfield-character-count-font-weight)
-	);
+	font-size: var(--mod-textfield-character-count-font-size, var(--spectrum-textfield-character-count-font-size));
+	font-family: var(--mod-textfield-character-count-font-family, var(--spectrum-textfield-character-count-font-family));
+	font-weight: var(--mod-textfield-character-count-font-weight, var(--spectrum-textfield-character-count-font-weight));
+	line-height: var(--mod-textfield-character-count-line-height, var(--spectrum-textfield-character-count-line-height));
 	grid-row: 1;
 	grid-column: 2 / span 1;
-	padding-inline-end: calc(
-		var(--mod-textfield-corner-radius, var(--spectrum-textfield-corner-radius)) /
-			2
-	);
-
-	.spectrum-Textfield--quiet & {
-		margin-block-end: var(
-			--mod-textfield-character-count-spacing-block-quiet,
-			var(--spectrum-textfield-character-count-spacing-block-quiet)
-		);
+	padding-inline-end: calc(var(--mod-textfield-corner-radius, var(--spectrum-textfield-corner-radius)) / 2);
```

Comment:
That's a good callout. I think I left this `padding-inline-end` here from before because there wasn't really any guidance one way or the other, but it doesn't look like there is any in the design.